### ISSUE TITLE
Acessing a secondary View (Stage)'s properties from a fragment get the primary ones.

### ISSUE THREAD

**Issue by jordaorodolfo** [edvin/tornadofx#769]
Hello! This is more a of a practical question than a bug report. Say that I have to views: `MainView` and `SecodaryView` and that at some point:

    class MainView {
        val secondaryView : SecondaryView by inject()
        fun openSec() {
            secondaryView.openWindow(modality = Modality.NONE)
        }
    }

and futher, say that I have a fragment `DetailFragment` embedded into this secondary View:

    class SecondaryView {
        val frag = DetailFragment()

        override val root = vbox {
            this += frag
        }       
    }

If I then try to get the properties of `currentStage`, `currentWindow` and `modalStage` from the `frag` `DetailFragment`, they all refer to the very first `MainView`'s open window, instead of the second one that just opened, in which the `frag` is embedded. Did I miss some very important from TFX architecture? Because, oddly enough, if from that `openSec` method on the `MainView` I call `secondaryView.currentStage?.requestFocus()`, it requests the focus correctly on the new opened window.

I appreciate any thoughts on the matter! Including "I think you'd better off with a code structure like X" for your purpose. The purpose BTW is just to open a more detailed view of a data model in traditional desktop fashion, by opening a detail window.

**Comment by edvin** [comment]
You must never manually instantiate UIComponent instances :) When you do so, you miss important life cycle events, and there is no way for the framework to attach the correct window to it. Try injecting the `DetailFragment` as well :)

You have two options. 

**1. Injection**

```kotlin
class SecondaryView {
    val frag: DetailFragment by inject()

    override val root = vbox {
        this += frag
    }       
}
```

**2. Add shortcut**

```kotlin
class SecondaryView {
    override val root = vbox {
        add<DetailFragment>()
    }       
}
```

**Comment by jordaorodolfo** [comment]
Shoot! That was so easy. Thanks a lot @edvin !! I just glanced again on the component chapter of the guide book at the embedding section and saw no mention of Fragments to be inject as well, although I think maybe some people would just extend the View logic to them, though I didn't ;). Would you mind just adding a line or so saying that fragments should ALSO be injected? In my code structure it somehow seemed more intuitive to do as I showed in this issue, but, had that small mention be there, I certainly would just have facepalmed myself and corrected everything! Thanks a lot man! Consider this corrected :)

**Comment by jordaorodolfo** [comment]
Actually, I've just followed your example and got a tiny error. You see, if I try to inject fragment `by inject`, it says that my `Fragment`s are not a subtype of `ScopedInstance`! Oh, and they are children fo TFX's `Fragment` class!
The second method works though, so I will be using it for the immediate moment, though I'd much rather use the injection route as it would need very few modifications in my code.

Specifically, the error given is

    Error:(27, 51) Kotlin: Type parameter bound for T in inline fun <reified T : Component> inject(overrideScope: Scope = ..., params: Map<String, Any?>? = ...): ReadOnlyProperty<Component, T> where T : ScopedInstance is not satisfied: inferred type AddressFragment is not a subtype of ScopedInstance

for the call
    
    val startAddressFragment : AddressFragment by inject(params = mapOf("addressModel" to serviceModel.startAddress, "label" to "Partida"))

being that the address class:

    class AddressFragment : Fragment() { ... }

Any lights on the matter? I can start a new issue if you wish!

All the best.

**Comment by jordaorodolfo** [comment]
I think it's best to move this to a new issue and closing this one, before we start a big converstion here.
